docs/configuration.md

```diff
@@ -60,15 +60,19 @@ Property
+-------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------
+**envoy-control.envoy.snapshot.incoming-permissions.enabled**                                                                                      | Enable incoming permissions                                                                                                                                                               | false
+**envoy-control.envoy.snapshot.incoming-permissions.client-identity-header**                                                                       | Header that identifies the client that called the endpoint                                                                                                                                | x-service-name
+**envoy-control.envoy.snapshot.incoming-permissions.source-ip-authentication.exact-ip-authentication.service-names**                               | Enable source ip based authentication for selected services                                                                                                                               | empty list
+**envoy-control.envoy.snapshot.incoming-permissions.source-ip-authentication.static-ip-authentication.ip-range-for-services**                      | Enable source ip based authentication for static ranges                                                                                                                                   | empty map
+**envoy-control.envoy.snapshot.incoming-permissions.source-ip-authentication.static-ip-authentication.ip-range-for-services.{name}**               | User chosen name for static ip range, it must match client name defined in incoming permissions                                                                                           | none
+**envoy-control.envoy.snapshot.incoming-permissions.source-ip-authentication.static-ip-authentication.ip-range-for-services.{name}.ip**            | Ip for static ip range                                                                                                                                                                    | empty string
+**envoy-control.envoy.snapshot.incoming-permissions.source-ip-authentication.static-ip-authentication.ip-range-for-services.{name}.prefix-length** | Prefix length of static ip range (eg in CIDR notation 192.168.1.0/24, 24 is the prefix and 192.168.1.0 is the ip)                                                                         | -1
```

Comment:
Not super important, but wouldn't this be more clear?

```
source-ip-authentication:
  enabled-for-services: ...
  match:
    service-names: "aaa", "bbb"
    ip-ranges:
      - "xxx": ...
      - "yyy": ...
```

Reply:
What would "enabled-for-services" mean in the second level declaration? Ip ranges do not have service name assigned it's more of a principal name that has to matched to metadata.

Reply:
The thing is, we have two separate mechanisms:
* source IP authentication based on service instances IP addresses from Consul
* source IP authentication based on statically defined IP ranges for defined service

IMO your config mix it and it may look like `enabled-for-services` is the main switch, which is not, because a service may be in `match` even if it is not in `enabled-for-services`. This config is similar to original one in this PR, and @lukidzi proposed to changed it for I think above reason.

I agree that this config is not super clear, maybe something like this will be better? LMK.

```
source-ip-authentication:
  ip-from-service-discovery: 
     enabled-for-services: ...
  static-ip-ranges:
     - service: <service name>
        ip-ranges: 
           - 1.2.3.4/24
           - 3.4.5.6/20
     - service: <service name>
        ip-ranges:
            - 3.5.6.3/12
```

Reply:
Ok, my bad, I thought `enabled-for-services` relates to both of the matching methods and means "which Envoys should have this feature enabled". Since it's not, my suggestion becomes:

```
source-ip-authentication:
  match:
    service-names: "aaa", "bbb"
    ip-ranges:
      - "xxx": ...
      - "yyy": ...
```

Your proposition @MarcinFalkowski also seems right to me. What I wanted to convey here is that "exact ip" vs "static ip" doesn't mean much unless you know the implementation. Configuring through properties should not require you to know the implementation.

Reply:
I think the current format is the same as your but some fields are renamed. Do I get this right?

Reply:
@pbetkier 
>Enable suggests to me that it's a feature flag to enable this mechanisms in the given services.
But it does this exactly - `enabled-for-services` enables feature `ip-from-service-discovery` for listed services

Reply:
With @pbetkier and @slonka we agreed to the following format:
```
source-ip-authentication:
  ip-from-service-discovery: 
     enabled-for-incoming-services: [serviceD, serviceE]
  ip-from-range:
     serviceZ:
       - 1.2.3.4/24
       - 3.4.5.6/20
     servicB: 
       - 3.5.6.3/12
```